examples/lock-app/p6/src/main.cpp

```diff
@@ -0,0 +1,123 @@
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <FreeRTOS.h>
+#include <mbedtls/threading.h>
+
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/KeyValueStoreManager.h>
+#include <support/CHIPMem.h>
+#include <support/CHIPPlatformMemory.h>
+
+#include <AppTask.h>
+
+#include "AppConfig.h"
+#include "DataModelHandler.h"
+#include "Server.h"
+#include "init_p6Platform.h"
+#if CHIP_ENABLE_OPENTHREAD
+#include <mbedtls/platform.h>
+#include <openthread/cli.h>
+#include <openthread/dataset.h>
+#include <openthread/error.h>
+#include <openthread/heap.h>
+#include <openthread/icmp6.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/platform/openthread-system.h>
+#include <openthread/tasklet.h>
+#include <openthread/thread.h>
+#endif // CHIP_ENABLE_OPENTHREAD
+
+using namespace ::chip;
+using namespace ::chip::Inet;
+using namespace ::chip::DeviceLayer;
+
+volatile int apperror_cnt;
+// ================================================================================
+// App Error
+//=================================================================================
+void appError(int err)
+{
+    P6_LOG("!!!!!!!!!!!! App Critical Error: %d !!!!!!!!!!!", err);
+    portDISABLE_INTERRUPTS();
+    while (1)
+        ;
+}
+
+// ================================================================================
+// FreeRTOS Callbacks
+// ================================================================================
+extern "C" void vApplicationIdleHook(void)
+{
+    // FreeRTOS Idle callback
+}
+
+// ================================================================================
+// Main Code
+// ================================================================================
+int main(void)
+{
+    int ret = CHIP_ERROR_SENTINEL;
+    init_p6Platform();
+
+    P6_LOG("==================================================\r\n");
+    P6_LOG("chip-p6-lock-example starting\r\n");
+    P6_LOG("==================================================\r\n");
+
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
+    // chip::DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().Init();
+
+    ret = PlatformMgr().InitChipStack();
+    if (ret != CHIP_NO_ERROR)
+    {
+        P6_LOG("PlatformMgr().InitChipStack() failed");
+        appError(ret);
+    }
+    chip::DeviceLayer::ConnectivityMgr().SetBLEDeviceName("P6_LOCK");
+    P6_LOG("Starting Platform Manager Event Loop");
+    ret = PlatformMgr().StartEventLoopTask();
+    if (ret != CHIP_NO_ERROR)
+    {
+        P6_LOG("PlatformMgr().StartEventLoopTask() failed");
+        appError(ret);
+    }
+    ret = GetAppTask().StartAppTask();
+    if (ret != CHIP_NO_ERROR)
+    {
+        P6_LOG("GetAppTask().Init() failed");
+        appError(ret);
+    }
+    /* Start the FreeRTOS scheduler */
+    vTaskStartScheduler();
+
+    chip::Platform::MemoryShutdown();
```

Comment:
Shouldn't this also StopEventLoopTask?  Or shouldn't something do it somewhere, at least?

Reply:
@bzbarsky-apple : platform shutdown is not called in this example and I dont see any call reference to StopEventLoopTask in other lock app examples. 
can you please review the updated changes and approve

Reply:
> platform shutdown is not called in this example and I dont see any call reference to StopEventLoopTask in other lock app examples.

Hmm.  The only way that we get to this MemoryShutdown call is if `vTaskStartScheduler` failed (i.e. we're out of memory).  Calling MemoryShutdown while we still leave the background task running seems odd.  But I guess the point is that if `vTaskStartScheduler` failed the background task is not running?  I just checked and it looks like `StopEventLoopTask` is a no-op (returns "not implemented") on FreeRTOS....

I guess this is fine, but we should clearly document why we are ok violating the documented PlatformManager API contract and not calling `StopEventLoopTask`.